master/internal/resourcemanagers/kubernetes_resource_manager.go

```diff
@@ -170,19 +176,28 @@ func (k *kubernetesResourceManager) Receive(ctx *actor.Context) error {
 
 func (k *kubernetesResourceManager) summarizeDummyResourcePool(
 	ctx *actor.Context,
-) *resourcepoolv1.ResourcePool {
+) (*resourcepoolv1.ResourcePool, error) {
 	slotsUsed := 0
 	for _, slotsUsedByGroup := range k.slotsUsedPerGroup {
 		slotsUsed += slotsUsedByGroup
 	}
 
+	resp := ctx.Ask(k.agent.handler, kubernetes.SummarizeResources{})
+	if err := resp.Error(); err != nil {
+		return nil, err
+	}
+	pods, ok := resp.Get().(*kubernetes.PodsInfo)
+	if !ok {
+		return nil, status.Error(codes.Internal, "unexpected response from actor")
+	}
+
 	return &resourcepoolv1.ResourcePool{
 		Name:                         kubernetesDummyResourcePool,
 		Description:                  "Kubernetes-managed pool of resources",
 		Type:                         resourcepoolv1.ResourcePoolType_RESOURCE_POOL_TYPE_K8S,
-		NumAgents:                    1,
+		NumAgents:                    int32(pods.NumAgents),
 		SlotType:                     k.config.SlotType.Proto(),
-		SlotsAvailable:               int32(k.agent.numSlots()),
+		SlotsAvailable:               int32(pods.NumSlots),
 		SlotsUsed:                    int32(k.agent.numUsedSlots()),
```

Comment:
this and a lot of the other fields that rely on `k.agent` are just borked. maybe we can follow up to this pr and fix them all?

Reply:
i guess @ioga will fix this by ripping `k.agent` out?

Reply:
well I was thinking when I rip them out, I'll just replace them with zeros. but yes it's be a good form to have such follow-up.